abm/connector.py

```diff
@@ -25,6 +26,16 @@ def __init__(self, config, logger, workdir):
         if 'connector' not in self.config:
             raise ValueError("'connector' field missing from configuration")
 
+        if 'vault_credentials' in self.config:
+            vault_credentials = self.config['vault_credentials']
+            del self.config['vault_credentials']
+            secrets = get_secrets_from_vault(vault_credentials=vault_credentials, datasetID=asset_name)
+            if secrets:
```

Comment:
maybe add a comment that not all connection types require creds. Later an error can be raised based on the connection type - for example for mysql where creds are required.

Reply:
In my opinion, the `airbyte-module` should not be aware of the different asset types. Ideally, it should not know which configuration fields are required for `mysql` (for example).